packages/grafana-ui/src/components/Collapse/CollapsableSection.tsx

```diff
@@ -1,55 +1,117 @@
-import React, { FC, ReactNode, useState } from 'react';
-import { css } from '@emotion/css';
+import React, { FC, ReactNode, useRef, useState } from 'react';
+import { uniqueId } from 'lodash';
+import { css, cx } from '@emotion/css';
 import { useStyles2 } from '../../themes';
-import { Icon } from '..';
+import { Icon, Spinner } from '..';
 import { GrafanaTheme2 } from '@grafana/data';
+import { getFocusStyles } from '../../themes/mixins';
 
 export interface Props {
   label: ReactNode;
   isOpen: boolean;
   /** Callback for the toggle functionality */
   onToggle?: (isOpen: boolean) => void;
   children: ReactNode;
+  className?: string;
+  contentClassName?: string;
+  loading?: boolean;
+  labelId?: string;
 }
 
-export const CollapsableSection: FC<Props> = ({ label, isOpen, onToggle, children }) => {
+export const CollapsableSection: FC<Props> = ({
+  label,
+  isOpen,
+  onToggle,
+  className,
+  contentClassName,
+  children,
+  labelId,
+  loading = false,
+}) => {
   const [open, toggleOpen] = useState<boolean>(isOpen);
   const styles = useStyles2(collapsableSectionStyles);
-  const headerStyle = open ? styles.header : styles.headerCollapsed;
   const tooltip = `Click to ${open ? 'collapse' : 'expand'}`;
-  const onClick = () => {
+  const onClick = (e: React.MouseEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+
     onToggle?.(!open);
     toggleOpen(!open);
   };
+  const { current: id } = useRef(uniqueId());
```

Comment:
Do we want to use `useId` instead here? https://react-spectrum.adobe.com/react-aria/useId.html